### ISSUE TITLE
Compile failure with Intel compilers

### ISSUE THREAD

**Issue by mdekstrand** [tidyverse/readxl#479]
I am trying to build the Tidyverse with the Intel C compiler, and `readxl` is failing to compile. From the error message, I suspect a missing header that `gcc` is more forgiving about.

---

Compile R with the Intel compiler; I followed the [instructions here](https://blog.ajsuarez.com/posts/compiling-r-icc-mkl/) and am using icc 18.0.1.

```
$ icc -v
icc version 18.0.1 (gcc version 4.8.5 compatibility)
```

```r
install.packages("readxl")
```

--- Error Message

```
icc -std=c99 -I"/home/mekstrand/opt/R-3.5.0-intel/lib64/R/include" -DNDEBUG -Iunix -I.  -I"/home/mekstrand/opt/R-3.5.0-intel/lib64/R/library/Rcpp/include" -I/usr/local/include   -fpic  -qopenmp -O3 -ipo -xHost -multiple-processes  -c endian.c -o endian.o
In file included from ./libxls/xlsstruct.h(40),
                 from libxls/endian.h(30),
                 from endian.c(33):
./libxls/ole.h(174): error: identifier "ssize_t" is undefined
  extern ssize_t ole2_read(void* buf,size_t size,size_t count,OLE2Stream* olest);
         ^

icc: error #10298: problem during post processing of parallel object compilation
compilation aborted for endian.c (code 2)
make: *** [endian.o] Error 2
ERROR: compilation failed for package ‘readxl’
```

**Comment by jennybc** [comment]
Do you think that this is somehow unique to readxl? Or are you perhaps learning some measures that someone would need to take, in general, to build a large set of R packages from source with `icc`? readxl has gotten an awful lot of use (mostly but not only via pre-built binaries) and this particular issue has never come up before.

FWIW, I note the error you show is, in fact, originating from the embedded libxls library, not code that is "native" to readxl. But obviously it should compile. I'm reluctant to make such changes in readxl itself, unless it's necessary for one of CRAN's official platforms. So perhaps this should just be documented somewhere.

**Comment by mdekstrand** [comment]
This isn't unique to readxl — I found a similar problem in r-lib/callr#50 (sent to r-lib/processx#109 - it turns out there that including the definitions in the source breaks Solaris builds). I am not sure if `readr` has the same problem (I put this in `Makevars` once I figured it out, and then was able to build the rest of the tidyverse).

In the entire `tidyverse` dependency tree, though, only `callr`, `readxl`, and possibly `readr` had a problem.

**Comment by jennybc** [comment]
> it turns out there that including the definitions in the source breaks Solaris builds

Yeah this is my exact concern about making changes in readxl itself.
